meteor/client/ui/Shelf/BucketPanel.tsx

```diff
@@ -919,3 +944,81 @@ export const BucketPanel = translateWithTracker<Translated<IBucketPanelProps>, I
 		)
 	)
 )
+
+interface DropzoneHolderProps {
+	id: string
+	bucketId: BucketId
+	url: string
+	hidden: boolean
+	t: TFunction<'translation', undefined>
+	showStyleBaseId: ShowStyleBaseId
+}
+const DropzoneHolder = (props: DropzoneHolderProps) => {
+	let frame: HTMLIFrameElement | null = null
+	let mounted = false
+
+	const setDropzoneElement = (frameRef: HTMLIFrameElement | null) => {
+		frame = frameRef
+		if (frame && !mounted) {
+			registerHandlers()
+			mounted = true
+		} else {
+			unregisterHandlers()
+			mounted = false
+		}
+	}
+	const registerHandlers = () => {
+		window.addEventListener('message', onMessage)
+	}
+	const unregisterHandlers = () => {
+		window.removeEventListener('message', onMessage)
+	}
+
+	const onMessage = (event: MessageEvent) => {
```

Comment:
```suggestion
	const onMessage = React.useCallback((event: MessageEvent) => {
```

I suggest wrapping this in a callback for reference stability.